README.md

```diff
@@ -43,36 +68,66 @@ An alternative is to install [Azure CLI](https://docs.microsoft.com/en-us/cli/az
    echo "---" >> cmd/clusterctl/examples/azure/out/provider-components.yaml
    kustomize build vendor/sigs.k8s.io/cluster-api/config/default/ >> cmd/clusterctl/examples/azure/out/provider-components.yaml
    ```
-3. Create the cluster. 
-   Kubernetes Version >= 1.11 is required to enable CRD subresources without needing a feature gate.
+3. Create the cluster.
 
+   **NOTE:** Kubernetes Version >= 1.11 is required to enable CRD subresources without needing a feature gate.
 
     **Linux**
     ```bash
-    ./bin/clusterctl create cluster --provider azure \
-    -m cmd/clusterctl/examples/azure/out/machines.yaml \
-    -c cmd/clusterctl/examples/azure/out/cluster.yaml \
-    -p cmd/clusterctl/examples/azure/out/provider-components.yaml \
-    --vm-driver kvm2 --minikube kubernetes-version=v1.12.2
+    ./bin/clusterctl create cluster -v 3 \
+    --provider azure \
+    --bootstrap-type kind \
+    -m ./cmd/clusterctl/examples/azure/out/machines.yaml \
+    -c ./cmd/clusterctl/examples/azure/out/cluster.yaml \
+    -p ./cmd/clusterctl/examples/azure/out/provider-components.yaml \
+    -a ./cmd/clusterctl/examples/azure/out/addons.yaml
     ```
 
     **macOS** 
     ```bash
-   ./bin/clusterctl create cluster --provider azure \
-   -m cmd/clusterctl/examples/azure/out/machines.yaml \
-   -c cmd/clusterctl/examples/azure/out/cluster.yaml \
-   -p cmd/clusterctl/examples/azure/out/provider-components.yaml \
-   --vm-driver virtualbox --minikube kubernetes-version=v1.12.2
+    ./bin/clusterctl create cluster -v 3 \
+    --provider azure \
+    --bootstrap-type kind \
+    -m ./cmd/clusterctl/examples/azure/out/machines.yaml \
+    -c ./cmd/clusterctl/examples/azure/out/cluster.yaml \
+    -p ./cmd/clusterctl/examples/azure/out/provider-components.yaml \
+    -a ./cmd/clusterctl/examples/azure/out/addons.yaml
    ```
+
 Once the cluster is created successfully, you can interact with the cluster using `kubectl` and the kubeconfig downloaded by the `clusterctl` tool.
 
 ```
-kubectl --kubeconfig=kubeconfig get clusters
-kubectl --kubeconfig=kubeconfig get machines
+export KUBECONFIG="$(kind get kubeconfig-path --name="clusterapi")"
+kubectl get clusters
+kubectl get machines
 ```
 
 ### Creating and using controller images
-TODO
 
-### Testing
-TODO
+Here's an example of how to build controller images, if you're interested in testing changes in the image yourself:
+
+```bash
+# Build the image.
+PREFIX=quay.io/k8s \
+NAME=cluster-api-azure-controller \
+TAG=0.2.0-alpha.3 \
+make docker-build
+
+# Push the image.
+PREFIX=quay.io/k8s \
+NAME=cluster-api-azure-controller \
+TAG=0.2.0-alpha.3 \
+make docker-push
+```
+
+**NOTE:** In order for the created images to be used for testing, you must push them to a public container registry.
```

Comment:
If this is the case, we should be exposing `REGISTRY` as a user configurable variable. (This can be done in another PR)

Reply:
This can already be configured with `PREFIX` and `NAME` in the Makefile.

Reply:
Although, maybe we could clean that up. I can assess it when I take a look at adding bazel and updating the Makefile.

Reply:
Sounds good.